homeworks/14/instagram-api/CommentServlet.java

```diff
@@ -0,0 +1,103 @@
+import java.sql.ResultSet;
+
+@WebServlet(name = "CommentServlet", value = "/posts/*")
+public class CommentServlet extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("text/html");
+        PrintWriter out = response.getWriter();
+        String pathInfo = request.getPathInfo(); // /{id}/comments
+        String[] pathParts = pathInfo.split("/");
+
+        String method;
+        if(pathParts.length == 2){
+            method = "post";
+        }
+        else {
+            method = "put";
+        }
+        out.println("<html><body>");
+        out.println("<form action=\"posts/*\" method=\""+ method +"\">");
+        out.println("comment: <input type=\"text\" name=\"comment\"><br>");
+        out.println("<input type=\"submit\" value=\"Submit\"></form>");
+        out.println("</body></html>");
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("text/html");
+        String pathInfo = request.getPathInfo(); // /{id}/comments
+        String[] pathParts = pathInfo.split("/");
+        String post_id = pathParts[1]; // {id}
+//      String part2 = pathParts[2]; // comments
+        String comment = request.getParameter("comment");
+        PrintWriter out = response.getWriter();
+
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+
+            Connection connection = DriverManager.getConnection("jdbc:mysql://localhost:3306/instagram", "root", "password");
+            PreparedStatement preparedStatement = connection.prepareStatement("insert into comments(username, comment, post_id) values(?,?,?)");
+            preparedStatement.setString(1,request.getSession().getAttribute("username").toString());
+            preparedStatement.setString(2,comment);
+            preparedStatement.setString(3,post_id);
+            preparedStatement.executeUpdate();
+
+            out.println(comment);
+        } catch (Exception e) {
+            out.print(e);
+        }
+
+    }
+
+    @Override
+    protected void doPut(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("text/html");
+
+        String pathInfo = request.getPathInfo(); // /{id}/comments//{id}
+        String[] pathParts = pathInfo.split("/");
+        String post_id = pathParts[1]; // {id}
+//      String part2 = pathParts[2]; // comments
+        String comment_id = pathParts[3]; // {id}
+
+        String comment = request.getParameter("comment");
+        PrintWriter out = response.getWriter();
+
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+
+            Connection connection = DriverManager.getConnection("jdbc:mysql://localhost:3306/instagram", "root", "password");

```

Comment:
You should close your database connection, prepared statement and results as they can cause memory leaks.